test/package/test_torchscript.py

```diff
@@ -486,6 +486,36 @@ def test_load_shared_tensors_repackaged(self):
             torch.allclose(loaded_mod_1.tensor, loaded_mod_1.sub_mod_1.tensor)
         )
 
+    def test_tensor_sharing_pickle(self):
+        """Test that saving a ScriptModule and a separately saving a tensor
+        object causes no issues.
+        """
+
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.foo = torch.ones(2, 3)
+
+            def forward(self):
+                return self.foo
```

Comment:
You could use 'ModWithTensor' from package_a.test_module if you wanted to and I think it would fill the need